agenda-services/src/main/java/org/exoplatform/agenda/rest/AgendaEventRest.java

```diff
@@ -675,6 +679,32 @@ public Response sendEventResponse(
     }
   }
 
+  @Path("search")
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(value = "Search the list of events available with query for an owner of type user or space, identified by its identity technical identifier."
+      + " If no designated owner, all events available for authenticated user will be retrieved.", httpMethod = "GET", response = Response.class, produces = "application/json")
+  @ApiResponses(value = { @ApiResponse(code = HTTPStatus.OK, message = "Request fulfilled"),
+      @ApiResponse(code = HTTPStatus.UNAUTHORIZED, message = "Unauthorized operation"),
+      @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error"), })
+  public Response searchEvent(@Context UriInfo uriInfo,
+                              @ApiParam(value = "Term to search", required = true) @QueryParam("query") String query,
+                              @ApiParam(value = "Offset", required = false, defaultValue = "0") @QueryParam("offset") int offset,
+                              @ApiParam(value = "Limit", required = false, defaultValue = "20") @QueryParam("limit") int limit) throws Exception {
+
+    offset = offset > 0 ? offset : RestUtils.getOffset(uriInfo);
+    limit = limit > 0 ? limit : RestUtils.getLimit(uriInfo);
+
+    String authenticatedUser = RestUtils.getCurrentUser();
+    Identity currentUser = CommonsUtils.getService(IdentityManager.class)
```

Comment:
you have already an IdentityManager here : 
https://github.com/exoplatform/agenda/blob/437c51febf497b636d8bd5518feb1ed2c5ff7ec0/agenda-services/src/main/java/org/exoplatform/agenda/rest/AgendaEventRest.java#L61

In addition, please use RestUtils class to retrieve current user information